velox/exec/SortWindowBuild.cpp

```diff
@@ -179,15 +181,37 @@ void SortWindowBuild::computePartitionStartRows() {
   partitionStartRows_.push_back(0);
 
   VELOX_CHECK_GT(sortedRows_.size(), 0);
-  for (auto i = 1; i < sortedRows_.size(); i++) {
-    if (partitionCompare(sortedRows_[i - 1], sortedRows_[i])) {
-      partitionStartRows_.push_back(i);
+  int32_t start = 0;
+  if (binarySearch_) {
+    while (start < sortedRows_.size()) {
+      int left = start;
```

Comment:
It might be cleaner to abstract the inner loop in a function called say findNextPartitionStartRow.

So 

```
while (start < sortedRows_.size()) {
  auto next = findNextPartitionStartRow(start);
  partitionStartRows_.push_back(next);
  start = next;
}
```

Reply:
done